src/avram/upsert.cr

```diff
@@ -0,0 +1,32 @@
+module Avram::Upsert
+  macro upsert_lookup_columns(*attribute_names)
+    def self.upsert!(*args, **named_args) : T
+      operation = new(*args, **named_args)
+      existing_record = find_existing_unique_record(operation)
+
+      if existing_record
+        operation.record = existing_record
```

Comment:
@wontruefree Ah, I think it's because here it would need to be public somehow. I guess an alternative could be doing something like `operation.set_record(_record: existing_record)`, but that probably just muddles up the code more than we save by just making it public and not documenting it. What do you think?

Reply:
I'll keep it public but not document. There may be other extensions/shards that need to hook into it. Having it there, but not documented should leave it open, but not leave regular users confused about what it is for

Reply:
That sounds good.